### PR TITLE
IMB-64, IMB-65 Upload UAN/DOB frontend

### PR DESCRIPTION
**Pull request by jamiecarterHO**
Relates [IMB-64](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-64) and [IMB-65](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-65)

## What?

Adding the form pages, validation and file submission for the current UAN/DOB upload solution.
* Adds a new app to the form 'uan'
* Adds Steps, views, translations and validations for the current solution design.
* Adds an `upload-csv` behaviour to handle validation errors and sending the file onwards.
* Adds CSV save method where the CSV file is uploaded via file-vault to the IMA S3 bucket. The returned FV URL is saved to its own table in the IMA RDS to be retrieved by the future processing container for out-of-hours processing of CSV
* There is a flag-enabled (config.js) option to save the CSV to disk and setup for a shared volume between the IMA and processing container.
* There will be a sibling PR for IMB-64 in the hof-rds-api repo that adds the new table to the current Knex DB schema migrations.

## Why?

There is no automated API to contact for the list of current UANs with their relevant dates of birth and other information for user verification, or to poll (e.g. nightly) for the list to store within our RDS.

As a tactical solution, the list (downloaded daily from the caseworking system as a CSV dump to be uploaded to the HOF IMA system) is not just additions but also deletions and potentially edits to RDS rows too. We will be supplied a full and cumulative canonical list with each upload which could grow to several thousand (~150,000 estimated max over 2 years) rows. 

As such we aim to process the upload of CSV rows into RDS rows in a separate container by cron job and during low traffic hours.

This PR adds the ability to upload a canonical CSV to S3, and save its retrieval URL for later processing. This is preferred to the backup option of using a Kubernetes shared volume to share the data between IMA and processing container because it is more robust, virus scans incoming CSV files, persists the data in case of pod or container failure and creates more of an audit trail.

## Testing

Unit tests are provided for validation methods in the `upload-csv` behaviour

## Anything else?

The ability to save the file to a shared volume remains but is switched off by flag in config.js. It may be removed if this backup functionality is eventually deemed unnecessary. It is probably useful to build this backup method into the processing service because some future forms that could make use of similar processing may not otherwise need or have a file-vault -> S3 setup as part of their architecture..

### CODE REVIEW

**Comment by jamiecarterHO** [comment]
hof-rds-api related PR: https://github.com/UKHomeOffice/hof-rds-api/pull/19

File: apps/uan/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+module.exports = {
+  name: 'uan',
+  baseUrl: '/uan',
+  steps: {
+    '/upload': {
+      behaviours: [UploadCSV],
+      fields: ['bulk-upload-uan', 'confirm-data-checked'],
+      next: '/complete'
+    },
+    '/complete': {
```
[the remark this answers is not in the document]
Comment (jamiecarterHO) [reply apps/uan/index.js:13]: No particular reason. Can update this to follow HOF convention.

**Comment by jamiecarterHO** [comment]
@Rhodine-orleans-lindsay updates made as per initial review and questions...

* `/complete` route and references is now `/confirmation` to go along with HOF convention
* Removed the code from `upload-csv` behaviour that saves files to shared volume as well as related config and creation of shared volume in the app deployment file
* Fixed some merge conflicts with newer stuff coming from the base branch.

**Review by Rhodine-orleans-lindsay — APPROVED** [review]
lgtm - really great work @jamiecarterHO !

**Pull request merged**